pkgs/tools/networking/tuic/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+let
+  buildTUIC = name: rustPlatform.buildRustPackage rec{
+    pname = "tuic";
+    version = "1.0.0";
+
+    src = fetchFromGitHub {
+      owner = "EAimTY";
+      repo = pname;
+      rev = "${name}-1.0.0";
+      sha256 = "sha256-VoNr91vDqBlt9asT/dwCeYk13UNiDexNNiKwD5DSn8k=";
+    };
+
+    cargoPatches = [
+      ./0001-add-Cargo.lock.patch
+    ];
```

Comment:
patches are hard to generate, just use `ln` as in `pkgs/tools/networking/dogdns/default.nix`